pkg/universal-cnf/vppagent/backend.go

```diff
@@ -54,6 +55,41 @@ func (b *UniversalCNFVPPAgentBackend) NewUniversalCNFBackend() error {
 	return nil
 }
 
+// CreateL2Memif runs the client/endpoint code for the VPP CNF to create vpp interface for layer 2 MEMIF
+func (b *UniversalCNFVPPAgentBackend) CreateL2Memif(dpConfig interface{}, ifName string, conn *connection.Connection, memifMaster bool) error {
+	vppconfig, ok := dpConfig.(*vpp.ConfigData) // type assertion
+	if !ok {
+		return fmt.Errorf("unable to convert dpconfig to vppconfig")
+	}
+
+	socketFilename := path.Join(getBaseDir(), memif.ToMechanism(conn.GetMechanism()).GetSocketFilename())
+
+	vppconfig.Interfaces = append(vppconfig.Interfaces,
+		&interfaces.Interface{
+			Name:    ifName,
+			Type:    interfaces.Interface_MEMIF,
+			Enabled: true,
+			// The MEMIF that communicates with client pod is master
+			// The MEMIF that communicates with the chain NSE pod is slave
+			Link: &interfaces.Interface_Memif{
+				Memif: &interfaces.MemifLink{
+					Master:         memifMaster,
+					SocketFilename: socketFilename,
+					RingSize:       512,
+				},
+			},
```

Comment:
What about `RxModes`? The `ProcessEndpoint` creates memif interface with `RxModes` set to:

```go
rxModes := []*interfaces.Interface_RxMode{
	{
		Mode:        interfaces.Interface_RxMode_INTERRUPT,
		DefaultMode: true,
	},
}
```